File: UI/lib/shared/models/requests/update_address_request.dart
class UpdateAddressDto {
  final String? street;
  final String? city;
  final String? postalCode;
  final String? country;

  UpdateAddressDto({
    this.street,
    this.city,
    this.postalCode,
    this.country,
  });

  Map<String, dynamic> toJson() {
    return {
      if (street != null) 'street': street,
      if (city != null) 'city': city,
      if (postalCode != null) 'postalCode': postalCode,
      if (country != null) 'country': country,
    };
  }
}

File: backend/TenderGo.Models/Requests/UpdateAddressRequest.cs
using System.ComponentModel.DataAnnotations;

namespace TenderGo.Models.DTOs
{
    public class UpdateAddressRequest
    {
        [MaxLength(200)]
        public string? Street { get; set; }

        [MaxLength(100)]
        public string? City { get; set; }

        [MaxLength(20)]
        public string? PostalCode { get; set; }

        [MaxLength(100)]
        public string? Country { get; set; }
    }
}